sos/plugins/sas3ircu.py

```diff
@@ -0,0 +1,47 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+
+# This sosreport plugin is meant for sas adapters.
+# This plugin logs inforamtion on each adapter it finds.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+import os
+
+
+class SAS3ircu(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+
+    plugin_name = "sas3ircu"
+
+    def setup(self):
+
+        # get list of adapters
+        output = self.call_ext_prog("sh -c 'sas3ircu list'", timeout=5)
+
```

Comment:
It's definitely not super common, but I can think off hand of at least on other plugin / family of plugins that could benefit from it - the old `emc` plugin was split into a bunch of smaller plugins a while back. They are broadly similar in terms of the class of data collected to your `sas3ircu` (i.e. they retrieve data from attached storage components, possibly using 3rd party / unpackaged tools provided by the vendor).

I think that's good enough justification to consider a generic mechanism - there are also possibly other plugins that I'm forgetting now that could be helped by this.